flow/FastAlloc.cpp

```diff
@@ -50,6 +54,9 @@
 #elif defined(__GNUG__)
 #ifdef __linux__
 #define INIT_SEG __attribute__ ((init_priority (1000)))
+#elif defined(__FreeBSD__)
+#pragma message "init_priority is not supported on this platform; will this be a problem?"
+#define INIT_SEG
```

Comment:
clang is officially supported in clang compiler. I compile success with __linux__ define on FreeBSD

Reply:
@ChasonTang Hmm ok, I'll take a look at this- this part of the code was brought in from ports but I know a lot is changing with C++ support and Boost in general. Thanks for the heads up.

Reply:
@ChasonTang alright this has been changed and haven't seen any problems with it so far at all. In the meantime I've updated my environment to FreeBSD 12 and have noticed a couple things.

TLS support only builds for me using QTCreator but not through the terminal. With CMake in the terminal for some reason it is looking in /usr/lib and not /usr/local/lib while linking causing an error. LibreSSL is found successfully using QTCreator though if you open CMakeLists.txt so I'm looking into that. 

fdbcli goes off into async land indefinitely if there is no fdbserver process or a db has not been configured. Start it with --no-status flag and make a db then it goes back to acting normal. Upgrading to 12 and this behavior may be mutually exclusive not sure yet.

Reply:
>fdbcli goes off into async land indefinitely

This was an independently noticed bug, which should be now fixed.